apps/console/src/features/identity-providers/components/forms/authenticators/apple-authenticator-form.tsx

```diff
@@ -0,0 +1,660 @@
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { Field, Form } from "@wso2is/form";
+import { Code, FormSection, GenericIcon, Hint } from "@wso2is/react-components";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, ReactNode, useEffect, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { Icon, SemanticICONS } from "semantic-ui-react";
+import { IdentityProviderManagementConstants } from "../../../constants";
+import {
+    AuthenticatorSettingsFormModes,
+    CommonAuthenticatorFormFieldInterface,
+    CommonAuthenticatorFormFieldMetaInterface,
+    CommonAuthenticatorFormInitialValuesInterface,
+    CommonAuthenticatorFormMetaInterface,
+    CommonAuthenticatorFormPropertyInterface,
+    CommonPluggableComponentMetaPropertyInterface
+} from "../../../models";
+
+/**
+ * Interface for Apple Authenticator Form props.
+ */
+interface AppleAuthenticatorFormPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * The intended mode of the authenticator form.
+     * If the mode is "EDIT", the form is used in the edit view and rely on metadata for readonly states, etc.
+     * If the mode is "CREATE", the form is used in the add wizards and all the fields will be editable.
+     */
+    mode: AuthenticatorSettingsFormModes;
+    /**
+     * Apple Authenticator metadata.
+     */
+    metadata: CommonAuthenticatorFormMetaInterface;
+    /**
+     * Apple Authenticator configured initial values.
+     */
+    initialValues: CommonAuthenticatorFormInitialValuesInterface;
+    /**
+     * Callback for form submit.
+     * @param values - Resolved Form Values.
+     */
+    onSubmit: (values: CommonAuthenticatorFormInitialValuesInterface) => void;
+    /**
+     * Is readonly.
+     */
+    readOnly?: boolean;
+    /**
+     * Flag to trigger form submit externally.
+     */
+    triggerSubmit: boolean;
+    /**
+     * Flag to enable/disable form submit button.
+     */
+    enableSubmitButton: boolean;
+    /**
+     * Flag to show/hide custom properties.
+     * @remarks Not implemented ATM. Do this when needed.
+     */
+    showCustomProperties: boolean;
+    /**
+     * Specifies if the form is submitting.
+     */
+    isSubmitting?: boolean;
+}
+
+/**
+ * Form initial values interface.
+ */
+interface AppleAuthenticatorFormInitialValuesInterface {
+    /**
+     * Apple Authenticator callback URL field value.
+     */
+    callbackUrl: string;
+    /**
+     * Apple Authenticator scopes field value.
+     */
+    Scopes: string;
+    /**
+     * Apple Authenticator client id field value.
+     */
+    ClientId: string;
+}
+
+/**
+ * Form fields interface.
+ */
+interface AppleAuthenticatorFormFieldsInterface {
+    /**
+     * Apple Authenticator additional query parameters field.
+     */
+    AdditionalQueryParams: CommonAuthenticatorFormFieldInterface;
```

Comment:
Any reason to start them with capital?

Reply:
We have defined property keys as this in backend